docs/configuration/query-frontend.md

```diff
@@ -0,0 +1,240 @@
+      #   memcached_client:
+      #     host: memcached.default.svc.cluster.local
+      #     service: memcached
+      #     consistent_hash: true
+
+frontend:
+  # 256 length tenant queues per frontend
+  max_outstanding_per_tenant: 256
+  log_queries_longer_than: 5s
+  compress_responses: true
+  
+frontend_worker:
+  address: query-frontend.<namespace>.svc.cluster.local:9095
+  grpc_client_config:
+    max_send_msg_size: 1.048576e+08
+  parallelism: 8
+```
+
+
+## Kubernetes Deployment
+
+Here's an example k8s deployment which can consume the above configuration yaml. Note that the above configuration snippet should be merged with the rest of your configuration and exposed as the `loki` configmap.
+
+### Frontend Service
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+  labels:
+    name: query-frontend
+  name: query-frontend
+  namespace: <namespace>
+spec:
+  ports:
+  - name: query-frontend-http-metrics
+    port: 80
+    protocol: TCP
+    targetPort: 80
+  - name: query-frontend-grpc
+    port: 9095
+    protocol: TCP
+    targetPort: 9095
+  selector:
+    name: query-frontend
+  sessionAffinity: None
+  type: ClusterIP
+```
+
+### Frontend Deployment
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  labels:
+    name: query-frontend
+  name: query-frontend
+  namespace: <namespace>
+spec:
+  minReadySeconds: 10
+  progressDeadlineSeconds: 600
+  replicas: 2
+  selector:
+    matchLabels:
+      name: query-frontend
+  strategy:
+    rollingUpdate:
+      maxSurge: 25%
+      maxUnavailable: 25%
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        name: query-frontend
+    spec:
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+          - labelSelector:
+              matchLabels:
+                name: query-frontend
+            # see https://kubernetes.io/docs/reference/kubernetes-api/labels-annotations-taints/#kubernetes-io-hostname
+            topologyKey: kubernetes.io/hostname
+      containers:
+      - args:
+        - -config.file=/etc/loki/config.yaml
+        - -log.level=debug
+        - -target=query-frontend
+        image: grafana/loki:<version>
+        imagePullPolicy: IfNotPresent
+        name: query-frontend
+        ports:
+        - containerPort: 80
+          name: http-metrics
+          protocol: TCP
+        - containerPort: 9095
+          name: grpc
+          protocol: TCP
+        resources:
+          limits:
+            memory: 1200Mi
+          requests:
+            cpu: "2"
+            memory: 600Mi
+        terminationMessagePath: /dev/termination-log
+        terminationMessagePolicy: File
+        volumeMounts:
+        - mountPath: /etc/loki
+          name: loki
+      restartPolicy: Always
+      terminationGracePeriodSeconds: 30
+      volumes:
+      - configMap:
+          defaultMode: 420
+          name: loki
+        name: loki
+```
+
+### Querier Deployment
```

Comment:
I wouldn't change the deployment of querier, I think we want this to be a quick way to test it.

I would just give frontend config,service and deployment, that can be added on top a working querier service.

Reply:
Good idea - simple is best.